packages/vite/src/node/plugins/define.ts

```diff
@@ -59,16 +59,18 @@ export function definePlugin(config: ResolvedConfig): Plugin {
       ...processEnv
     }
 
+    const characters = '[\'"`\\/]'
+
+    const replacementsStr = Object.keys(replacements)
+      .map((str) => {
+        return str.replace(/[-[\]/{}()*+?.\\^$|]/g, '\\$&')
+      })
+      .join('|')
+
     const pattern = new RegExp(
       // Do not allow preceding '.', but do allow preceding '...' for spread operations
-      '(?<!(?<!\\.\\.)\\.)\\b(' +
-        Object.keys(replacements)
-          .map((str) => {
-            return str.replace(/[-[\]/{}()*+?.\\^$|]/g, '\\$&')
-          })
-          .join('|') +
-        // prevent trailing assignments
-        ')\\b(?!\\s*?=[^=])',
+      `(?<!${characters})(?<!(?<!\\.\\.)\\.)\\b(${replacementsStr})\\b(?!${characters})(?!\\s*?=[^=])`,
+      // prevent trailing assignments
```

Comment:
I think we should try to split this regex to multiple lines so the comments match, and would be more readable too.